src/leap/mx/mail_receiver.py

```diff
@@ -378,10 +385,32 @@ def _step_process_mail_backend(self, filepath):
                 defer.returnValue(None)
 
             log.msg("Encrypting message to %s's pubkey" % (uuid,))
-            doc = yield self._encrypt_message(pubkey, mail_data)
+            try:
+                doc = yield self._encrypt_message(pubkey, mail_data)
+
+                yield self._export_message(uuid, doc)
+                yield self._remove(filepath)
+            except Exception as e:
+                yield self._bounce_with_timeout(filepath, msg, e)
+
+    @defer.inlineCallbacks
+    def _bounce_with_timeout(self, filepath, msg, error):
+        if filepath not in self._bounce_timeout:
+            self._bounce_timeout[filepath] = datetime.now()
+            log.msg("New staled email {0!r}: {1!r}".format(filepath, error))
+            defer.returnValue(None)
 
-            yield self._export_message(uuid, doc)
-            yield self._remove(filepath)
+        bounce_date = datetime.now() - timedelta(days=self.BOUNCE_TIMEOUT_DAYS)
+        if bounce_date > self._bounce_timeout[filepath]:
+            log.msg("Bouncing staled email {0!r}: {1!r}"
+                    .format(filepath, error))
+            bounce_reason = "There was a problem in the server and the " \
+                            "email could not be delivered."
+            yield self._bounce_message(msg, filepath, bounce_reason)
+        else:
+            stalled_days = datetime.now() - self._bounce_timeout[filepath]
```

Comment:
if you assign the current now() to a var in the block above, no need to evaluate it again here
same for the bounce_start_timestamp, I think it'll improve readabilty